pkg/content/iterator/html.go

```diff
@@ -0,0 +1,167 @@
+package iterator
+
+import (
+	"strings"
+
+	"github.com/andybalholm/cascadia"
+	"github.com/pkg/errors"
+	"github.com/readium/go-toolkit/pkg/content/element"
+	"github.com/readium/go-toolkit/pkg/fetcher"
+	"github.com/readium/go-toolkit/pkg/manifest"
+	"github.com/readium/go-toolkit/pkg/mediatype"
+	"golang.org/x/net/html"
+	"golang.org/x/net/html/atom"
+)
+
+type HTMLContentIterator struct {
+	resource        fetcher.Resource
+	locator         manifest.Locator
+	BeforeMaxLength int // Locators will contain a `before` context of up to this amount of characters.
```

Comment:
This property doesn't need to be public, but we should be able to set it when calling `NewHTML()` or using `HTMLFactory`.

```suggestion
	beforeMaxLength int // Locators will contain a `before` context of up to this amount of characters.
```

Reply:
@mickael-menu Unfortunately because of the language constraints, we can't have an optional parameter to a function with a default value like what is done in the kotlin-toolkit. I think making the property public is acceptable because I imagine a situation like this, which is also used in other go libraries:
```go
// direct usage
x := iterator.NewHTML(resource, locator)
x.BeforeMaxLength = 123
x.HasNext()

// or based on the current ContentElements() func
it := content.Iterator()
it.(*iterator.HTMLContentIterator).BeforeMaxLength = 123
hasNext, err := it.HasNext()
```
This is the "substitute" for that language feature. I personally think requiring the beforeMaxLength to be specified in the initial function is "ugly", because it is something that will probably not be changed in typical usage, which is why the kotlin-toolkit makes it optional.

Reply:
That's true, and I don't think we can customize it on the Kotlin toolkit anyways, unless you create an `HtmlResourceContentIterator` manually.